examples/systemcalls/systemcalls.c

```diff
@@ -92,8 +131,50 @@ bool do_exec_redirect(const char *outputfile, int count, ...)
  *   The rest of the behaviour is same as do_exec()
  *
 */
+    bool ret_status = true;
+
+    int fd = open(outputfile, O_WRONLY|O_TRUNC|O_CREAT, 0644);
 
+    if (fd < 0)
+    {
+        printf("Failed to open the file %s\n", outputfile);
+        return false;
+    }
+
+    pid_t pid = fork();
+
+    if (pid == -1)
+    {
+        printf("Error: Could not create child process\n");
+        ret_status = false;
+    }
+    else if (pid == 0)
+    {
+        if (dup2(fd, 1) < 0)
+        {
+            perror("dup2");
+            ret_status = false;
+        }
+        close(fd);
+
+        int status = execv(command[0], command);
+
+        if (status != 0) {
+            printf("Error: Could not execute the \"%s\" command\n", command[0]);
+            exit(EXIT_FAILURE);
+        }
+    }
+    else {
+        int cmd_status;
+
+        if(waitpid(pid, &cmd_status, 0) == -1 )
+            ret_status = false;
+        if (WEXITSTATUS(cmd_status))
+            ret_status = false;
```

Comment:
good! 
See discussion in class about avoiding duplication between these two very similar functions.